Description: add missing include to fix build failure
Author: Dmitry Shachnev <mitya57@ubuntu.com>
Forwarded: https://codereview.qt-project.org/103387
Last-Update: 2015-01-12

--- a/src/organizer/qorganizerrecurrencerule.h
+++ b/src/organizer/qorganizerrecurrencerule.h
@@ -43,6 +43,7 @@
 #define QORGANIZERRECURRENCERULE_H
 
 #include <QtCore/qdatetime.h>
+#include <QtCore/qmetatype.h>
 #include <QtCore/qset.h>
 #include <QtCore/qshareddata.h>
