=== modified file 'src/versit/qversitreader_p.cpp'
Index: qtpim/src/versit/qversitreader_p.cpp
===================================================================
--- qtpim.orig/src/versit/qversitreader_p.cpp
+++ qtpim/src/versit/qversitreader_p.cpp
@@ -42,6 +42,7 @@
 #include <QtCore/qbuffer.h>
 #include <QtCore/qtextcodec.h>
 #include <QtCore/qvariant.h>
+#include <QtCore/QReadWriteLock>
 
 #include "qversitutils_p.h"
 
@@ -427,7 +428,18 @@ QVersitReaderPrivate::~QVersitReaderPriv
 
 QHash<QPair<QVersitDocument::VersitType,QString>, QVersitProperty::ValueType>*
 QVersitReaderPrivate::valueTypeMap() {
+    static QReadWriteLock valueTypeMapLock;
+
+    QReadLocker readLock(&valueTypeMapLock);
     if (mValueTypeMap == 0) {
+        readLock.unlock();
+
+        QWriteLocker writeLock(&valueTypeMapLock);
+        // check if map still empty
+        if (mValueTypeMap) {
+            return mValueTypeMap;
+        }
+
         mValueTypeMap = new QHash<QPair<QVersitDocument::VersitType,QString>, QVersitProperty::ValueType>();
         mValueTypeMap->insert(qMakePair(QVersitDocument::VCard21Type, QString::fromLatin1("AGENT")),
                              QVersitProperty::VersitDocumentType);
